File: src/main/java/ru/nsu/fit/team_project/model/commands/CreateObjectCommand.java
package ru.nsu.fit.team_project.model.commands;

import ru.nsu.fit.team_project.model.MObject;
import ru.nsu.fit.team_project.model.Model;

import java.util.UUID;

public class CreateObjectCommand extends Command {
    private final UUID objectID;
    private final String objectType;

    public CreateObjectCommand(UUID objectID, String objectType) {
        super("CreateObjectCommand");
        this.objectID = objectID;
        this.objectType = objectType;
    }

    @Override
    public void execute(Model model) {
        model.addObject(new MObject(objectID, objectType));
    }
}

File: src/jvmMain/kotlin/data/model/commands/CreateObjectCommand.kt
package data.model.commands

import data.model.Model
import data.model.ModelObject
import java.util.UUID

class CreateObjectCommand(
    private val objectId: UUID,
    private val objectType: String
) : Command(COMMAND_NAME) {

    private companion object {

        private const val COMMAND_NAME = "CreateObject"
    }

    override fun execute(model: Model) {
        model.addObject(
            obj = ModelObject(
                id = objectId,
                type = objectType
            )
        )
    }
}